CMakeLists.txt

```
@@ -334,6 +336,72 @@ if(URIPARSER_BUILD_TESTS)
     add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND})
 endif()
 
+#
+# Fuzzers
+#
+if(URIPARSER_BUILD_FUZZERS)
+    if(NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
+        message(FATAL_ERROR
+            "Building fuzz targets without Clang (but ${CMAKE_C_COMPILER_ID}) is not "
+            "supported. Please set -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++.")
+    endif()
+
+    if(URIPARSER_OSSFUZZ_BUILD AND NOT DEFINED ENV{LIB_FUZZING_ENGINE})
+        message(FATAL_ERROR
+            "OSS-Fuzz builds require the environment variable "
+            "LIB_FUZZING_ENGINE to be set. If you are seeing this "
+            "warning, it points to a deeper problem in the ossfuzz "
+            "build setup.")
+    endif()
+
+    add_library(fuzzparser STATIC ${API_HEADER_FILES} ${LIBRARY_CODE_FILES})
+    target_compile_definitions(fuzzparser PUBLIC URI_STATIC_BUILD)
+    target_include_directories(fuzzparser
+        PUBLIC
+            ${CMAKE_CURRENT_SOURCE_DIR}/include
+        PRIVATE
+            ${CMAKE_CURRENT_BINARY_DIR}  # for UriConfig.h
+    )
+
+    if(NOT URIPARSER_OSSFUZZ_BUILD)
+        target_compile_options(fuzzparser PRIVATE -fsanitize=fuzzer-no-link)
+    endif()
+
+    function (add_fuzzer target_name source compile_definitions)
+        add_executable(${target_name} ${source})
+        target_link_libraries(${target_name} PRIVATE fuzzparser)
+        target_compile_definitions(${target_name} PRIVATE ${compile_definitions})
+        if(URIPARSER_OSSFUZZ_BUILD)
+            set_target_properties(${target_name} PROPERTIES LINK_FLAGS $ENV{LIB_FUZZING_ENGINE})
+            set_target_properties(${target_name} PROPERTIES LINKER_LANGUAGE "CXX")
+        else()
+            target_compile_options(${target_name} PRIVATE -fsanitize=fuzzer-no-link)
+            set_target_properties(${target_name} PROPERTIES LINK_FLAGS -fsanitize=fuzzer)
+        endif()
+    endfunction()
```

```suggestion
        endif()
        set_property(TARGET ${target_name} PROPERTY RUNTIME_OUTPUT_DIRECTORY fuzz)        
    endfunction()
```
(It's a detail, but would be nice. Matches Expat.)

---

Added